adapters/get_character_controller_test.go

```diff
@@ -0,0 +1,103 @@
+package adapters
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"testing"
+
+	"github.com/SergioSanchezWizeline/academy-go-q12021/domain/model"
+	"github.com/SergioSanchezWizeline/academy-go-q12021/interfaces"
+	"github.com/SergioSanchezWizeline/academy-go-q12021/mocks"
+	"github.com/SergioSanchezWizeline/academy-go-q12021/usecase"
+	"github.com/gorilla/mux"
+)
+
+func TestNewGetCharacterController(t *testing.T) {
+	type args struct {
+		usecase usecase.GetCharacterUserCase
+	}
+	tests := []struct {
+		name string
+		args args
+		want interfaces.Controller
+	}{
+		{
+			"Nrew controller success",
+			args{&mocks.GetCharacterUserCase{}},
+			&getCharacterController{&mocks.GetCharacterUserCase{}},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := NewGetCharacterController(tt.args.usecase); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("NewGetCharacterController() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_getCharacterController_ServeHTTP(t *testing.T) {
+	tests := []struct {
+		name     string
+		id       int
+		expected string
+		wantErr  bool
+	}{
+		{
+			"Get character success 1",
+			1,
+			`{"id":1,"name":"Name"}`,
+			false,
+		},
+		{
+			"Get character success 2",
+			2,
+			`{"id":2,"name":"Name"}`,
+			false,
+		},
+		{
+			"Should raise error",
+			0,
+			``,
+			true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+
+			usecase := &mocks.GetCharacterUserCase{}
+			if tt.wantErr {
+				usecase.On("Execute", tt.id).Return(nil, errors.New("error"))
```

Comment:
Same question about the `Execute` function.

Reply:
Same